### ISSUE TITLE
Cursor skips forwards

### ISSUE THREAD

**Issue by apblack** [gracelang/grace-web-editor#9]
After typing a number of "cursor-forward" characters, the cursor goes crazy and starts skipping wildly.

**Comment by KimBruce** [comment]
I noticed that as well.

**Comment by apblack** [comment]
Seems to be fixed now.  Perhaps a new version of ACE fixed it for us.  In any case, I can no longer reproduce it.

**Comment by apblack** [comment]
Here is a program that reproduces it:
```
import "objectdraw" as od
method named(name) {
  object {
    var origin is readable := 0@0
  }
}
```
Put the cursor in front of `var`, and move the cursor right with the arrow keys.  It gets to the front to `origin`, and then skips to the end of the line.

**Comment by KimBruce** [comment]
The same problem illustrates the off-by-one error with the cursor under Safari.

Put the cursor in the middle of 0@0 and try adding a space separating the @ from the 0's.  It works fine on Chrome, but not Safari (which perhaps we don't care about).

**Comment by Dmitri-2** [comment]
Fixed by commit b964e0a.

**Comment by apblack** [comment]
This is caused by setting the tab width to be "4" rather than 4, i.e., a string rather than a number.
The code is attempting to make the cursor skip forward to the next tab stop.  In JavaScript, adding, for example,  28 + "4" produces "284", which is then converted to 284 — usually past the end of the line!

The solution is to convert the tab wide to a number when it is read from local storage.
